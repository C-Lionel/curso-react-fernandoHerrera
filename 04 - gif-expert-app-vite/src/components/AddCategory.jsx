import React, { useState } from 'react';

export const AddCategory = ( { onNewCategory } ) => {


    const [inputValue, setInputValue] = useState('');

    const handleInputChange = ({target}) => {
        setInputValue(target.value)
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length <= 1) return; 
        // setCategories( categories => [...categories, inputValue]);
        onNewCategory(inputValue.trim())
        setInputValue('');
    }

    

    return (

            <form onSubmit={ handleSubmit }>

                <input
                    type= 'text'
                    placeholder= 'Buscar gifs'
                    value= { inputValue }
                    // onChange = {(event) => handleInputChange(event) } ==== > se puede hacer asi o de la otra manera porque el primer argumento es igual primer argumento que se envia en la funciÃ³n
                    onChange= { handleInputChange }
    
                 />

            </form>

        
    )
}
