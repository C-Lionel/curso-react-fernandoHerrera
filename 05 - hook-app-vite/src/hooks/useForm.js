import { useState } from "react";

export const useForm = ( initialForm = {} ) => {

    const [formState, setFormState] = useState( initialForm );

    const onInputChange = ({ target }) => {

        const { name, value } = target;
        setFormState({
            ...formState,
            [name]: value // toma el valor del name que puede ser username o email y le agrega el value
        });

    }

    const onResetForm = () => {
        setFormState(initialForm)
    }

    return {
        ...formState,
        formState,
        onInputChange,
        onResetForm
    }

}
