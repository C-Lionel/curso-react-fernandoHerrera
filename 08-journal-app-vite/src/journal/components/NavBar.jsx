import { LogoutOutlined, MenuOutlined } from "@mui/icons-material";
import { AppBar, Grid, IconButton, Toolbar, Typography, List, Drawer, Divider, Box } from "@mui/material";
import { useDispatch, useSelector } from "react-redux";




import { startLogout } from "../../store/auth";
import { SideBarItem } from "./SideBarItem";


export const NavBar = () => {

    const dispatch = useDispatch()

    const onLogout = () => {
        dispatch(startLogout())
    }

    const { displayName } = useSelector(state => state.auth);
    const { notes } = useSelector(state => state.journal);

    return (
        <AppBar position="fixed">
            <Toolbar>

                <IconButton
                    color="inherit"
                    edge="start"
                >


                    <button className="btn btn-light" type="button" data-bs-toggle="offcanvas" data-bs-target="#staticBackdrop" aria-controls="staticBackdrop">
                        <MenuOutlined />
                    </button>

                    <div className="offcanvas offcanvas-start w-100" data-bs-backdrop="static" tabindex="-1" id="staticBackdrop" aria-labelledby="staticBackdropLabel">
                        <div className="offcanvas-header">
                            <h5 className="offcanvas-title" id="staticBackdropLabel"></h5>
                            <button type="button" className="btn-close m-1" data-bs-dismiss="offcanvas" wi aria-label="Close"></button>
                        </div>
                        <div className="offcanvas-body">

                            <div>


                                <Box
                                    component='nav'
                                    sx={{ flexShrink: { sm: 0 } }}
                                >

                                    <Drawer
                                        data-bs-dismiss="offcanvas"
                                        variant="permanent"
                                        open={true}
                                        sx={{
                                            display: { xs: 'block' },
                                            '& .MuiDrawer-paper': { boxSizing: 'border-box', width: '80%' }
                                        }}
                                    >

                                        <Toolbar>
                                            <Typography variant="h6" noWrap component='div'>
                                                {displayName}
                                            </Typography>
                                        </Toolbar>

                                        <Typography variant="h6" noWrap component='div'>
                                            NOTAS
                                        </Typography>

                                        <Divider />

                                        <List >
                                            {
                                                notes.map(note => (
                                                    <SideBarItem key={note.id} {...note} />
                                                ))
                                            }
                                        </List>

                                    </Drawer>

                                </Box>

                            </div>
                        </div>
                    </div>
                </IconButton>

                <Grid container direction='row' justifyContent='space-between' alignItems='center'>
                    <Typography variant="h6" noWrap component='div'> JournalApp </Typography>
                    <IconButton
                        
                        color="error"
                        onClick={onLogout}
                    >
                        <LogoutOutlined style={{ fontSize: 40 }} />
                    </IconButton>
                </Grid>

            </Toolbar>

        </AppBar>
    )
}
