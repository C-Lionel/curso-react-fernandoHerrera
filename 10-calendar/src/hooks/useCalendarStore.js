import { useDispatch, useSelector } from "react-redux"
import { onAddNewEvent, onDeleteEvent, onSetActiveEvent, onUpdateEvent } from "../store";


export const useCalendarStore = () => {

    const dispatch = useDispatch()

    const { events, activeEvent } = useSelector(state => state.calendar);

    const setActiveEvent = (calendarEvent) => {
        dispatch(onSetActiveEvent(calendarEvent))
    }

    const startSavingEvent = async (calendarEvent) => {
        // Llegar al backend
        // Todo sale bien
        if (calendarEvent._id) {
            // Actualizando
            dispatch(onUpdateEvent({ ...calendarEvent }))
        } else {
            // Creando
            dispatch(onAddNewEvent({ ...calendarEvent, _id: new Date().getTime() })) // Lo desestructura porque es el payload de mi reducer
        }
    }

    const startDeletingEvent = () => {
        // Llegar al backend
        dispatch( onDeleteEvent() )
    }



    return {
        //* Propiedades

        events,
        activeEvent,
        hasEventSelected: !!activeEvent?._id, // La doble negación (!!) hace que si activeEvent es null va a regresar false y si tiene un objeto va a regresar true

        //* Métodos
        setActiveEvent,
        startSavingEvent,
        startDeletingEvent

    }

}
