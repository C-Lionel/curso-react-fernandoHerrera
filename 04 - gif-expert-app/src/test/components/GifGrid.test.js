import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');



describe('Pruebas en el componente <GifGrid />', () => {

    const category = 'One Punch'


    test('El componente <GifGrid /> debe de mostrarse correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={category} />);

        expect(wrapper).toMatchSnapshot();

    })

    test('Debe de mostrar items cuando se cargan imÃ¡genes useFetchGifs', () => {

        const gifs = [
            {
                id: 'ABC',
                url: 'https://google.com.ar',
                title: 'Google'
            },
            {
                id: '123',
                url: 'https://google.com.ar',
                title: 'Google'
            }

        ];



        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={category} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);

    })



})
