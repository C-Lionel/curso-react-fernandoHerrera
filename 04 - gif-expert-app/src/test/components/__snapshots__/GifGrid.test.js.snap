// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifGrid /> Debe de mostrar items cuando se cargan im√°genes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="category-h3 animate__animated animate__bounceIn"
  >
     
    One Punch
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Google"
      url="https://google.com.ar"
    />
    <GifGridItem
      id="123"
      key="123"
      title="Google"
      url="https://google.com.ar"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en el componente <GifGrid /> El componente <GifGrid /> debe de mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="category-h3 animate__animated animate__bounceIn"
  >
     
    One Punch
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
     Loading... 
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
